sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -227,13 +228,14 @@ case class AtomicReplaceTableAsSelectExec(
 
   override protected def doExecute(): RDD[InternalRow] = {
     val schema = query.schema.asNullable
+    val propertiesWithOwner = withDefaultOwnership(properties).asJava
```

Comment:
can we add an analyzer rule to catch all the CREATE TABLE like commands and add the ownership properties?

Reply:
do we have any other APIs will bypass the analyzer?

Reply:
If we only look at the logical plan, I think only the following commands need to be considered:
1. CREATE TABLE
2. CREATE TABLE AS SELECT
3. REPLACE TABLE
4. REPLACE TABLE AS SELECT

Reply:
Yes, they have a common parent `V2CreateTablePlan` when they are on the logical side. We may add a method like 
```scala
override def withPartitioning(rewritten: Seq[Transform]): V2CreateTablePlan = {
    this.copy(partitioning = rewritten)
}

override def withOwnership(): V2CreateTablePlan = {
    this.copy(properties = this.properites ++ ownership)
}
```

but not more elegant and simpler as it is now

Reply:
a logical plan may have multiple physical plans, e.g. the atomic version of CTAS. That's why I think it's better to do it at the logical phase.

Reply:
maybe we can add them in DataSourceV2Strategy? then we don't need extra analyzer rules to copy themselves

Reply:
that also works